tests/StateSnapshot.test.js

```diff
@@ -235,6 +261,85 @@ const Component = React.memo(({props})=>{
      `,
       errors: [PROXY_RENDER_PHASE_MESSAGE, PROXY_RENDER_PHASE_MESSAGE],
     },
+    {
+      code: `
+          const state = proxy({ height: 0})
+          export const Component = () => {
+            return (
+              <div>
+                <button style={{ height: state.height }}>abc</button>
+              </div>
+            )
+          }
+          `,
+      errors: [PROXY_RENDER_PHASE_MESSAGE],
+    },
+    {
+      code: `
+          const state = proxy({ array: [1,2,3]})
+          export const Component = () => {
+            return (
+              <div>
+                <CustomComponent options={state.array}>abc</CustomComponent>
+              </div>
+            )
+          }
+          `,
+      errors: [PROXY_RENDER_PHASE_MESSAGE],
+    },
+    {
+      code: `
+          const state = proxy({ show: false })
+          export const Component = () => {
+            return (
+              <div>
+              {
+                state.show && <CustomComponent>abc</CustomComponent>
+              }
+              </div>
+            )
+          }
+          `,
+      errors: [PROXY_RENDER_PHASE_MESSAGE],
+    },
+    {
+      code: `
+          const state = proxy({ array: [1,2,3]})
+          export const Component = () => {
+            const snapshot = useSnapshot(state)
+            const filtered = snapshot.array.filter((item) => item !== 1)
+            return (
+              <CustomComponent data={filtered} />
+            )
+          }
+          `,
+      errors: [SNAPSHOT_CALLBACK_MESSAGE],
```

Comment:
For this one, There's no derived value checks in place, so i'm not sure we wanna add this test in place right now. 
@dai-shi

Reply:
I might be missing something, but why is this listed in invalid case?

Reply:
I reviewed it again. I think this should be a valid case. We cannot trigger re-render if we will use state in this case.